File: qlty-cli/tests/lang/cpp/basic.in/BooleanLogic.cpp
#include <iostream>

class BooleanLogic {
public:
    bool foo;
    bool bar;
    bool baz;
    bool qux;
    bool zoo;
    bool woo;

    void f0() {
        int x = 1 - 2 + 3;
    }

    void f1() {
        if (foo && bar && baz && qux && zoo && woo) {
            return;
        }
    }

    void f2() {
        auto check = [this]() {
            return foo || bar || baz || qux || zoo || woo;
        };
        if (check()) {
            std::cout << "lambda check passed" << std::endl;
        }
    }
};

File: qlty-cli/tests/lang/vbnet/basic.in/BooleanLogic.vb
Public Class BooleanLogic
    Private foo As Integer
    Private bar As Integer

    Private Sub F0()
        Dim x = foo - bar + 1
    End Sub
End Class

Public Class BooleanLogic1
    Private foo As Boolean
    Private bar As Boolean
    Private baz As Boolean
    Private qux As Boolean
    Private zoo As Boolean

    Private Sub F1()
        If foo AndAlso bar AndAlso baz AndAlso qux AndAlso zoo Then
            Return
        End If
    End Sub
End Class
